src/Admin/AbstractAdmin.php

```diff
@@ -3262,24 +3266,33 @@ public function canAccessObject($action, $object)
     }
 
     /**
-     * @return string[]
+     * @return MutableTemplateRegistryInterface
      */
-    protected function configureExportFields(): array
+    final public function getTemplateRegistry()
     {
-        return $this->getModelManager()->getExportFields($this->getClass());
+        if ($this->hasTemplateRegistry()) {
+            return $this->templateRegistry;
+        }
+
+        throw new \LogicException(sprintf('Unable to find the templateRegistry for admin `%s`', static::class));
```

Comment:
This is not really BC, before I get null, now I get a LogicException. Also I would change the order, because on other methods we first check if not has, then throw exception (or deprecation in some cases in 3.x) and the if it has, return it.

Reply:
I add `trigger_error` and comment Exception as next_major. Now this PR should be acceptable.